lib/cocoapods-core/source/aggregate.rb

```diff
@@ -198,26 +187,34 @@ def update_search_index(search_data)
       # @param  [Set] set
       #         The set for which the information is needed.
       #
-      # @note   If the specification can't load an empty hash is returned and
+      # @note   If the specification can't load an empty array is returned and
       #         a warning is printed.
       #
       # @note   For compatibility with non Ruby clients a strings are used
       #         instead of symbols for the keys.
       #
-      # @return [Hash{String=>String}] A hash with the search information.
+      # @return [Array<String>] An array of words contained by the set's search related information.
       #
-      def search_data_from_set(set)
-        result = {}
+      def word_list_from_set(set)
         spec = set.specification
-        result['version'] = spec.version.to_s
-        result['summary'] = spec.summary
-        result['description'] = spec.description
-        result['authors'] = spec.authors.keys.sort * ', '
-        result
+        word_list = [set.name.dup]
+        if spec.summary
+          word_list += spec.summary.split
```

Comment:
I agree with you on moving tokenization to a later issue to get things moving.




I just assumed you were doing an intersection - which is why I suggested using symbols in the inverted index array values (for memory savings and more efficiency on the intersection).

On Fri, Oct 23, 2015 at 1:18 AM, Muhammed Yavuz Nuzumlali
<notifications@github.com> wrote:

>>          spec = set.specification
>> -        result['version'] = spec.version.to_s
>> -        result['summary'] = spec.summary
>> -        result['description'] = spec.description
>> -        result['authors'] = spec.authors.keys.sort * ', '
>> -        result
>> +        word_list = [set.name.dup]
>> +        if spec.summary
>> +          word_list += spec.summary.split
> I will definitely want to challenge cleverness of the current system with better tokenization methods, but I feel like it should be another enhancement issue, and we should go with current implementation.
> I guess I detected a more critical issue. Current implementation won't work for multi word queries like this: `pod search --full network library`.
> I should split query string to array and perform search for each query, and then get the intersect of the pod arrays. It will take some time
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/CocoaPods/Core/pull/265/files#r42818410